contrib_bots/lib/giphy/giphy.py

```diff
@@ -0,0 +1,128 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# ~/.giphy_config
+
+from __future__ import absolute_import
+from __future__ import print_function
+from six.moves.configparser import SafeConfigParser
+import requests
+import logging
+import sys
+import os
+
+def parse_giphy_config(config_file):
+    config = SafeConfigParser()
+    config.readfp(f, config_file)
+    global giphy_api_key
```

Comment:
I would avoid using a global variable here. You know the name of the config key right? So you can lookup that when you need it.